api/wallet.go

```diff
@@ -15,6 +16,37 @@ type WalletSiafundsBalance struct {
 	SiacoinClaimBalance types.Currency
 }
 
+// scanAmount scans a types.Currency.
+func scanAmount(amountStr string) (amount types.Currency, err error) {
+	// exponential format
+	if strings.ContainsAny(amountStr, "Ee") {
+		amountRat := new(big.Rat)
+		_, err = fmt.Sscan(amountStr, amountRat)
+		if err != nil {
+			return
+		}
+		amount = types.NewCurrency(new(big.Int).Div(amountRat.Num(), amountRat.Denom()))
+		return
+	}
+
+	// standard format
+	_, err = fmt.Sscan(amountStr, &amount)
+	return
+}
+
+// scanAddres scans a types.UnlockHash.
+func scanAddress(addrStr string) (addr types.UnlockHash, err error) {
+	var addrBytes []byte
+	_, err = fmt.Sscanf(addrStr, "%x", &addrBytes)
+	if err != nil {
+		return
+	}
+	if copy(addr[:], addrBytes) != len(addr) {
+		err = errors.New("Malformed coin address")
+	}
+	return
+}
```

Comment:
maybe a type should be added to the return values. 'this is a siacoin addr' 'this is a siafund addr', etc.